File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.js
// Last updated: 4/29/2026, 10:36:24 AM
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
    const map = {};
    let read = 0;
    let write = 0;
    while(read < nums.length) {
        const val = nums[read];
        if(map[val] === undefined) {
            map[val] = true;
            if(write < read) {
                nums[write] = val;    
            }
            write++;
        }
        read++;
    }
    return write;
};

File: 26-RemoveDuplicatesFromSortedArray/26-RemoveDuplicatesFromSortedArray.py
# Last updated: 4/29/2026, 10:36:30 AM
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        prev = None
        while i < len(nums):
            curr = nums[i]
            if prev != None:
                if prev == curr:
                    nums.pop(i)
                    continue
            prev = curr
            i += 1
        return i